lib/elixir/test/elixir/kernel/cli_test.exs

```diff
@@ -96,10 +96,28 @@ defmodule Kernel.CLI.CompileTest do
   test "fails on missing patterns" do
     fixture = fixture_path "compile_sample.ex"
     output = elixirc('#{fixture} non_existing.ex -o #{tmp_path}')
-    assert :string.str(output, 'non_existing.ex') > 0, "expected non_existing.ex to be mentionned"
-    assert :string.str(output, 'compile_sample.ex') == 0, "expected compile_sample.ex to not be mentionned"
+    assert :string.str(output, 'non_existing.ex') > 0, "expected non_existing.ex to be mentioned"
+    assert :string.str(output, 'compile_sample.ex') == 0, "expected compile_sample.ex to not be mentioned"
     refute File.exists?(tmp_path("Elixir.CompileSample.beam")), "expected the sample to not be compiled"
   end
+
+  test "fails on missing write access to .beam file" do
```

Comment:
Let's replace this by an unit test? We don't need to go through elixirc, let's just try to call the parallel compiler. I hope it will be faster and that we can keep the test async. :)